http/middleware.go

```diff
@@ -188,11 +188,30 @@ func AuthorizedPartyMatches(parties ...string) func(string) bool {
 	}
 }
 
-// CustomClaims allows to pass a type (e.g. struct), which will be populated with the token claims based on json tags.
-// You must pass a pointer for this option to work.
-func CustomClaims(claims any) AuthorizationOption {
+// CustomClaimsConstructor allows to pass a constructor function
+// which returns a pointer to a type (struct) to hold custom token
+// claims.
+// The instance of the custom claims type will be then made available
+// through the clerk.SessionClaims struct.
+//
+//	// Define a type to describe the custom claims.
+//	type MyCustomClaims struct {
+//		ACustomClaim string `json:"a_custom_claim"`
+//	}
+//
+//	// In your HTTP server mux, configure the middleware with
+//	// the custom claims constructor.
+//	WithHeaderAuthorization(CustomClaimsConstructor(func() any {
+//		return &MyCustomClaims{}
+//	})
+//
+//	// In the HTTP handler, access the active session claims. The
+//	// custom claims are available in the SessionClaims.Custom field.
+//	sessionClaims, ok := clerk.SessionClaimsFromContext(r.Context())
+//	customClaims, ok := sessionClaims.Custom.(*MyCustomClaims)
+func CustomClaimsConstructor(constructor func() any) AuthorizationOption {
```

Comment:
❓ Would it make sense to pass the request context in the constructor?

```suggestion
func CustomClaimsConstructor(constructor func(ctx context.Context) any) AuthorizationOption {
```

Reply:
Yes, but since this goes down to `jwt.Verify`, the context might not make sense there. 

We can have different params for middleware and `jwt.Verify` though. I'll revise this.

Reply:
I was wrong, jwt.Verify does accept a context. :smile: 

Updated.

Reply:
Mostly noted this because it's going to be tough to change afterwards.

Reply:
And you were 100% right to do so :pray: :smile: